tests/Contrib/Unit/ZipkinSpanConverterTest.php

```diff
@@ -176,6 +181,54 @@ public function shouldConvertAnOTELSpanOfUnknownKindToAZipkinSpanOfUnspecifiedKi
         $this->assertArrayNotHasKey('kind', $row);
     }
 
+    /**
+     * @test
+     */
+    public function shouldConvertAnEventWithoutAttributesToAnAnnotationWithOnlyItsName()
+    {
+        $span = (new SpanData())
+            ->addEvent('event.name', new Attributes());
```

Comment:
Updated, though not sure I fully understand what this is for tbth (closest I can imagine is to make tests more unit-y so failures are easier to pinpoint? Or something similar? Would be curious to understand this a bit better). Also not entirely sure what an easy way is to know if it worked

Reply:
Code coverage normally records all code which has been **executed** during a test, whether it has been explicitly tested or not. So when dependecies are not mocked, it will create false postives in code coverage. In other words in this case code coverage is not a good metric anymore to see what parts of the code has been covered by unit tests. (It's not a good metric to show _what_ has been tested anyway, only what has _not_ been tested.)

In order to avoid these false positives in PHPUnit there are basically tow ways. Either you mock away _all_ dependencies of the tested library, or you tell PHPUnit with the '@covers' annotation, what part of the code you actually trying to test.

You can execute phpunit like this to create a HTML view of the code coverage: 
```bash
vendor/bin/phpunit --coverage-html=tests/coverage/html
```
You can find the HTML code coverage report in `tests/coverage/html`

(I have created a PR (https://github.com/open-telemetry/opentelemetry-php/pull/477) to add a make target to create the report)

Reply:
Maybe one more thing regarding mocks vs. covers annotation. 

A lot of people do not like writing mocks, which is understandable since they can get very tricky. Especially when you deal with a lot of nested dependencies and have to replicate a lot of the behavior of the mocked classes, it can get complicated.

However , I prefer writing mocks for one reason. If I find myself writing a lot of complicated mocks, it can be an indicator for a subpar design choice I did (for example "violating" [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle) or [LoD](https://en.wikipedia.org/wiki/Law_of_Demeter) ) . So the solution is refactoring the code to be less complicated, and the mocks will be less complicated as well. 
(Needless to say, I only know this, because I made a lot of bad design choices in the past)

Reply:
Yup I think that makes sense to me, making the code coverage data better highlight which units aren't being directly, intentionally tested vs having them only being incidentally indirectly covered via other tests (is how I'm conceptualizing it now at least).

And yeah I feel that too, when mocking gets really complicated I've felt before the real problem is often elsewhere and I'm just working around it.